rir/src/compiler/opt/elide_env.cpp

```diff
@@ -26,6 +26,7 @@ void ElideEnv::apply(RirCompiler&, ClosureVersion* function, LogStream&) const {
                     if (!envIsNeeded) {
                         i->elideEnv();
                         i->type.setNotObject();
+                        i->effects.reset(Effect::ExecuteCode);
```

Comment:
this looks good, but if we do that , we should probably rename the `envOnlyForObj` instruction. because right now the function name only suggests that no env is needed. someone might be tempted to set this to true for an instruction that still executes code, but just doesn't need an env...